core/pta/device.c

```diff
@@ -53,6 +56,25 @@ static TEE_Result get_devices(uint32_t types,
 		}
 	}
 
+	if (IS_ENABLED(CFG_EARLY_TA)) {
+		const struct early_ta *eta;
+
+		for_each_early_ta(eta) {
+			if (!(eta->flags & TA_FLAG_DEVICE_ENUM))
+				continue;
+			if (ip_size < sizeof(TEE_UUID)) {
+				res = TEE_ERROR_SHORT_BUFFER;
+			} else {
+				tee_uuid_to_octets(uuid_octet, &eta->uuid);
+				memcpy(device_uuid, uuid_octet,
+				       sizeof(TEE_UUID));
+				device_uuid++;
+				ip_size -= sizeof(TEE_UUID);
+			}
+			op_size += sizeof(TEE_UUID);
+		}
```

Comment:
Minor suggestion: for code to be easier to read out (and maintain), may I suggest to following sequence:
```c
		for_each_early_ta(eta) {
			if (!(eta->flags & TA_FLAG_DEVICE_ENUM))
				continue;

			op_size += sizeof(TEE_UUID);

			if (ip_size < sizeof(TEE_UUID)) {
				res = TEE_ERROR_SHORT_BUFFER;
				continue;
			}

			tee_uuid_to_octets(uuid_octet, &eta->uuid);
			memcpy(device_uuid, uuid_octet, sizeof(TEE_UUID));
			device_uuid++;
			ip_size -= sizeof(TEE_UUID);
		}

Reply:
well,  this file has 2 about the same cycles. I can write it differently here but cycle for pseudo also needed to be rewrote. I prefer separate patch for that then correct this one.